### ISSUE TITLE
pkgconfig include dir prevents redis plus plus from building

### ISSUE THREAD

**Issue by delleceste** [redis/hiredis#1044]
Hello.
installing hiredis with the following configuration:

cmake .. -DCMAKE_INSTALL_PREFIX=/opt/hiredis

results in a pkgconfig file like this in /opt/hiredis/lib/pkgconfig/

```
prefix=/opt/hiredis
install_libdir=lib
exec_prefix=${prefix}
libdir=${exec_prefix}/${install_libdir}
includedir=${prefix}/include
pkgincludedir=${includedir}/hiredis

Name: hiredis
Description: Minimalistic C client library for Redis.
Version: 1.0.3
Libs: -L${libdir} -lhiredis
Cflags: -I${pkgincludedir} -D_FILE_OFFSET_BITS=64

```
 redis-plus-plus configured  and later installed with

cmake .. -DCMAKE_INSTALL_PREFIX=/opt/redis-plus-plus -DCMAKE_PREFIX_PATH=/opt/hiredis

builds correctly as well

But when using pkgconfig to build a third party software relying on both redis and redis plus plus I see that redis-plus-plus includes 

*<hiredis/hiredis.h>

which breaks compilation because hiredis pkg config file sets 

  *-I/opt/hiredis/include/hiredis*

instead of *-I/opt/hiredis/include*

Changing  *pkgincludedir=${includedir}/hiredis* to  *pkgincludedir=${includedir}* fixes the issue.

Thanks for your attention.

Giacomo

**Comment by bjosv** [comment]
I will prepare a PR for this. I have a small additional fix for another finding, but will build-test it a lot first..

**Comment by bjosv** [comment]
I have looked into this a bit now and it seem that we don't really have a consistent view how to use the library and its include paths. Maybe we need to settle this before changing for the pkg-config.

The variants I have seen so far are:
* CMakes package via `hiredis-config.cmake` gives `hiredis_INCLUDEDIR "${PACKAGE_PREFIX_DIR}/include"`
  and the user can preferable use `#include <hiredis/hiredis.h>` when using the installed package.

* redis.com shows an example using `#include <hiredis/hiredis.h>`
  https://developer.redis.com/develop/C/

but at the same time:

* The README.md mentions `#include "hiredis.h"`
  https://github.com/redis/hiredis#using-it

* The examples in `<repo>/examples/` uses `#include <hiredis.h>`
In a bugreport (regarding pkg-config in Debians hiredis package) the examples are showcased as how to define the includes:
 https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=739834

My view is that `#include <hiredis/hiredis.h>` is commonly preferred, which means that we should update the pkg-config conf,
but in that case  we should update the Readme and examples as well.

Any thoughts @michael-grunder, et al.

**Comment by michael-grunder** [comment]
> My view is that #include <hiredis/hiredis.h> is commonly preferred, which means that we should update the pkg-config conf, but in that case, we should update the Readme and examples as well.

Agreed.  I have always viewed `#include <hiredis/hiredis.h>` as canonical.

We should update `pkg-config` and the docs, but I assume our examples are including `#include <hiredis.h>` because it's much simpler to do it that way when building hiredis (e.g. we don't need the headers installed already).

In terms of `pkg-config` I suppose the big question is whether the modifications can be done in a nonbreaking way.
